rmgpy/solver/base.pyx

```diff
@@ -988,25 +1006,40 @@ cdef class ReactionSystem(DASx):
                 for i in xrange(numCoreSpecies):
                     if not unimolecularThreshold[i]:
                         # Check if core species concentration has gone above threshold for unimolecular reaction
-                        if coreSpeciesConcentrations[i] > unimolecularThresholdVal:
-                            unimolecularThreshold[i] = True
+                        if fluxBasis == 'mass':
+                            if coreSpeciesConcentrations[i]*MWW[i] > unimolecularThresholdVal:
```

Comment:
I think it would look cleaner and reduce code duplication to define a `fluxFactor` which is 1 when `fluxBasis='mole'`.
```
fluxFactor = self.MWW[i] if fluxBasis == 'mass' else 1.0
if coreSpeciesConcentrations[i] * fluxFactor > unimolecularThresholdVal:
    unimolecularThreshold[i] = True
```

And similar setup for bi and trimolecular.

Reply:
I've tried to clean it up and remove most of the if statements, the fluxFactor approach just seems wasteful in the loop to me, although it probably doesn't slow us down that much since I don't think the filter loop is limiting here.

Reply:
Considering the readability, @mliu49's suggestion makes sense. The current if statement seems a little bit too long for me.

Reply:
I think the revised if statement is ok. Could you add a line break though?